pint_xarray/accessors.py

```diff
@@ -540,12 +540,133 @@ def reindex_like(
             fill_value=fill_value,
         )
 
+    def interp(
+        self,
+        coords=None,
+        method="linear",
+        assume_sorted=False,
+        kwargs=None,
+        **coords_kwargs,
+    ):
+        """unit-aware version of interp
+
+        Just like :py:meth:`xarray.DataArray.interp`, except the object's indexes are converted
+        to the units of the indexers first.
```

Comment:
this doesn't look entirely correct, but since it is copied from `reindex` I'll open a new PR for this